pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -178,3 +177,83 @@ func (i *deploymentClientImpl) listDeploymentsForConfig(namespace, configName st
+		// If the latest deployment exists and is complete, clear the dc message
+		return "", existingDeployments, latestDeploymentExists, nil
+	}
+
+	// Rest of the code will handle non-existing or failed latest deployment causes
+	// TODO: Inspect pod logs in case of failed latest
+
+	// If the configuration we are inspecting doesn't have a config change trigger
+	// there is no reason in proceeding to inspect its image change trigger
+	if !deployutil.HasChangeTrigger(config) {
+		return "", existingDeployments, latestDeploymentExists, nil
+	}
+
+	details := ""
+	// Look into image change triggers and find out possible deployment failures such as
+	// missing image stream tags with or without build configurations pointing at them
+	for _, trigger := range config.Triggers {
+		if trigger.Type != deployapi.DeploymentTriggerOnImageChange || trigger.ImageChangeParams == nil || !trigger.ImageChangeParams.Automatic {
+			continue
+		}
+		name := trigger.ImageChangeParams.From.Name
+		tag := trigger.ImageChangeParams.Tag
+
+		// Check if the image stream tag pointed by the trigger exists
+		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
+			if !errors.IsNotFound(err) {
+				return details, existingDeployments, latestDeploymentExists, err
+			}
+			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
+			// (a build configuration output points to it so it's going to be populated at some point in the
+			// future)
+			details = fmt.Sprintf("The config trigger for %q will have no effect because \"%s:%s\" does not exist", deployutil.LabelForDeploymentConfig(config), name, tag)
+			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+			if err != nil {
+				return details, existingDeployments, latestDeploymentExists, err
+			}
+
+			for _, bc := range bcList.Items {
+				if bc.Spec.Output.To != nil && bc.Spec.Output.To.Kind == "ImageStreamTag" {
+					parts := strings.Split(bc.Spec.Output.To.Name, ":")
+					if len(parts) != 2 {
+						return details, existingDeployments, latestDeploymentExists, fmt.Errorf("invalid image stream tag: %s", bc.Spec.Output.To.Name)
+					}
+					if parts[0] == name && parts[1] == tag {
+						details = fmt.Sprintf("\"%s:%s\" does not exist yet but will be created at some point since buildConfig %q points to it", name, tag, bc.Name)
+						break
+					}
+				}
+			}
+
+		}
+	}
+	return details, existingDeployments, latestDeploymentExists, nil
+}
+
+// updateDetails updates a deployment configuration with the provided details
+func (c *DeploymentConfigController) updateDetails(config *deployapi.DeploymentConfig, details string) error {
+	var err error
+	if config.Details == nil {
+		config.Details = new(deployapi.DeploymentDetails)
+	}
+	if details != config.Details.Message {
+		config.Details.Message = details
+		_, err = c.osClient.DeploymentConfigs(config.Namespace).Update(config)
```

Comment:
Return the updated resource and re-assign it at the call-site so that any subsequent code that wants to make more updates to the config won't get a resource conflict.